### DISCUSSION TITLE
How to Create an Image Slider Like Microsoft Store?

### DISCUSSION THREAD

**Discussion by ghost1372** [microsoft/microsoft-ui-xaml#9325] [Q&A] [answered]
Hi, @Sergio0694
since you are working on MS Store can you help me a little?
i want to create an image slider something similar to microsoft store, i know that MS Store uses FlipView + PipsPager but some things are unknown to me, for example:
What kind of effect is used for the images? (when hovering the mouse over the images)
Every image has a mask right? (OpacityMask?)
How do the images look together? I can only show one picture.
![Screenshot 2024-02-09 123118](https://github.com/microsoft/microsoft-ui-xaml/assets/9213496/e3103f4a-560f-474e-9126-8efc1d93edb5)
![Screenshot 2024-02-09 123223](https://github.com/microsoft/microsoft-ui-xaml/assets/9213496/733c6ad4-4209-45ab-9a32-01ecdd4901d0)

If I'm not mistaken, you have already open sourced some features of the store, is it possible to share these items as well?

**Comment by JimmyRespawn** [comment]
I think you can create this with 5 flipviews side by side. In the bottom I think it's PipsPager[WinUI3 and 2].
But I think the store team might create this with custom controls.

**Comment by Sergio0694** [comment, marked as answer]
We're not using a `FlipView`, that collection is a fully custom, virtualized, `ItemsRepeater`-based control with a whole bunch of custom composition animations and interaction behaviors, built by @ryken100. I'm not sure if we have plans to open source it shortly though.

**Comment by ghost1372** [comment]
I’ve created a control similar to the Store Carousel (a high-copy :) ). You can use it in your projects.
https://github.com/ghost1372/DevWinUI

![](https://raw.githubusercontent.com/ghost1372/DevWinUI-Resources/refs/heads/main/DevWinUI-Docs/StoreCarousel.gif)
